src/agents/memory-optimizer/memory-optimization-agent.ts

```
@@ -0,0 +1,857 @@
+
+  /**
+   * Determine entity type from entity ID or action metadata
+   */
+  private determineEntityType(entityId: string, action: any): string {
+    // Try to determine from action metadata first
+    if (action.entityType) {
+      return action.entityType;
+    }
+
+    // Try to determine from entity ID prefix
+    if (entityId.startsWith('comp-')) return 'component';
```

ESLint

[error] Expected { after 'if' condition.

(curly)